src/transformState.js

```diff
@@ -5,7 +5,32 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+
+      case 'addProperties':
+        for (const key in extraData) {
+          state[key] = extraData[key];
+        }
+        break;
+
+      case 'removeProperties':
+        const toRemove = keysToRemove;
+
+        for (let j = 0; j < toRemove.length; j++) {
+          delete state[toRemove[j]];
+        }
```

Comment:
don't see any sense in using  for loop, better to use` for of` or `forEach` here